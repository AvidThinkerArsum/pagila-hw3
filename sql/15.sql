/*
 * Find every documentary film that is rated G.
 * Report the title and the actors.
 *
 * HINT:
 * Getting the formatting right on this query can be tricky.
 * You are welcome to try to manually get the correct formatting.
 * But there is also a view in the database that contains the correct formatting,
 * and you can SELECT from that VIEW instead of constructing the entire query manually.
 */ 


SELECT title, STRING_AGG(
        UPPER(SUBSTRING(first_name FROM 1 FOR 1)) || LOWER(SUBSTRING(first_name FROM 2)) ||

        UPPER(SUBSTRING(last_name FROM 1 FOR 1)) || LOWER(SUBSTRING(last_name FROM 2)), ', '
    ) AS actors
FROM (
SELECT a.actor_id, a.first_name, a.last_name, r.title, r.film_id, r.rating FROM actor a
JOIN lateral (
    SELECT film_id, title, rating FROM film_actor
    JOIN film USING (film_id)
    JOIN film_category USING (film_id)
    JOIN category USING (category_id)
    WHERE actor_id = a.actor_id
    AND rating = 'G' AND name = 'Documentary'

) r ON TRUE
) AS t
GROUP BY title
ORDER BY title;
