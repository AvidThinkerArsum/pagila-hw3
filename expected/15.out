         title          |                                                                                       actors                                                                                        
------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 AFFAIR PREJUDICE       | JodieDegeneres, OprahKilmer, ScarlettDamon, KennethPesci, FayWinslet
 ANNIE IDENTITY         | CateMcqueen, GretaKeitel, AdamGrant
 BREAKFAST GOLDFINGER   | RussellTemple, JuliaZellweger, MichaelBolger, AlGarland, AngelinaAstaire
 BUTTERFLY CHOCOLAT     | BurtTemple, NickStallone, MenaTemple, KimAllen, LisaMonroe, EdGuiness, MaryKeitel
 CASABLANCA SUPER       | AngelaWitherspoon, KirstenAkroyd, WalterTorn, ReeseWest
 CITIZEN SHREK          | KevinBloom, IanTandy, BenHarris, RussellTemple, KennethTorn, CaryMcconaughey, JudeCruise, GaryPhoenix
 COLDBLOODED DARLING    | BenWillis, GinaDegeneres, BetteNicholson, LucilleTracy
 COLOR PHILADELPHIA     | SalmaNolte, DarylCrawford, ChristopherWest, OlympiaPfeiffer, JohnSuvari, PenelopeGuiness, GrouchoDunst
 CONNECTION MICROCOSMOS | NatalieHopkins, RussellTemple, VivienBasinger, FayWinslet, ReneeTracy, ScarlettDamon, CameronStreep
 CRUELTY UNFORGIVEN     | WillWilson, GregChaplin
 DARN FORRESTER         | RalphCruz, SandraKilmer, BobFawcett, MenaHopper, CateMcqueen, ChrisBridges, SeanGuiness, CharlizeDench, JodieDegeneres
 DRACULA CRYSTAL        | NickWahlberg, JodieDegeneres, CarmenHunt, CameronWray, MichelleMcconaughey, SeanWilliams, BenWillis, GregChaplin, MorganHopkins, DarylCrawford, MorganWilliams, IanTandy, ReeseWest
 GRAPES FURY            | LucilleDee, MaryTandy, HenryBerry, CarmenHunt, BurtTemple
 HAPPINESS UNITED       | AdamGrant, NickWahlberg, MerylGibson, CubaAllen, GrouchoDunst
 JUMANJI BLADE          | GaryPhoenix, JimMostel, JenniferDavis, BobFawcett, NickStallone, MenaTemple
 LAWLESS VISION         | MichaelBolger, PenelopePinkett, BobFawcett, JoeSwank
 LOVER TRUMAN           | VivienBergen, ChristianNeeson, LisaMonroe, MenaHopper, MillaKeitel, ThoraTemple, ReeseKilmer
 MEET CHOCOLATE         | GrouchoWilliams, WarrenNolte, ParkerGoldberg, TimHackman, ZeroCage, WarrenJackman
 MOONWALKER FOOL        | RayJohansson, WoodyHoffman, UmaWood, ZeroCage, MaryKeitel, HumphreyGarland, DarylCrawford
 NOVOCAINE FLIGHT       | BenWillis, KarlBerry, MichaelBolger, RussellTemple, WalterTorn, GregChaplin
 SHINING ROSES          | SusanDavis, JuliaBarrymore, HumphreyGarland, CateHarris, WoodyJolie
 SIDE ARK               | AudreyOlivier, ChrisDepp, LaurenceBullock, JayneSilverstone, HelenVoight, JadaRyder
 SWEETHEARTS SUSPECTS   | JayneNolte, JuliaMcqueen, HelenVoight, JoeSwank, BenWillis
 TORQUE BOUND           | NatalieHopkins, AdamHopper, GeneHopkins, KennethHoffman, UmaWood
(24 rows)

